### PR TITLE
apache arrow 9.0.0

### PR DESCRIPTION
**Pull request by jeroen**

### CODE REVIEW

**Comment by jeroen** [comment]
Looking good.

File: Formula/apache-arrow.rb [hunk 1]
```diff
@@ -40,6 +37,8 @@ def install
       -DARROW_USE_GLOG=OFF
       -DARROW_PYTHON=OFF
       -DARROW_S3=ON
+      -DARROW_WITH_BROTLI=ON
```
Comment (nealrichardson) [inline Formula/apache-arrow.rb:41]: Comparing to https://github.com/apache/arrow/blob/master/dev/tasks/homebrew-formulae/autobrew/apache-arrow.rb, we also should have this on (though up above, sorted alphabetically; I just can't suggest up there)

```suggestion
      -DARROW_GCS=ON
      -DARROW_WITH_BROTLI=ON
```

Comment (nealrichardson) [reply Formula/apache-arrow.rb:41]: (Likewise for the other variations on this formula)

Comment (jeroen) [reply Formula/apache-arrow.rb:41]: What is gcs? Does this not need any extra deps?

Comment (nealrichardson) [reply Formula/apache-arrow.rb:41]: google cloud storage. google-cloud-cpp and its dependencies (notably abseil-cpp) get built in the Arrow C++ build if they're not found, as with any other dependencies. We could certainly set up an autobrew formula for it, or for abseil, but it's not strictly required, it just makes the arrow build take a few extra minutes.

**Review by assignUser — APPROVED** [review]
We add `-DARROW_VERBOSE_THIRDPARTY_BUILD=ON` for better CI troubleshooting but that shouldn't be added here. Other than that I don't see anything missing. (except the alphabetical order of the cmake args ;) ) 👍

**Pull request merged**